### PR TITLE
move stuff to Details namespace

### PR DESCRIPTION
**Pull request by Jinming-Hu**

### CODE REVIEW

**Comment by katmsft** [comment]
@vhvb1989 This test case is unstable, can you please take a look? 
```
71: [  FAILED  ] TransportAdapter.connectionPoolTest (191604 ms)
71: [----------] 1 test from TransportAdapter (191604 ms total)
71: 
71: [----------] Global test environment tear-down
71: [==========] 1 test from 1 test suite ran. (191604 ms total)
71: [  PASSED  ] 0 tests.
71: [  FAILED  ] 1 test, listed below:
71: [  FAILED  ] TransportAdapter.connectionPoolTest
71: 
71:  1 FAILED TEST
 71/114 Test  #71: azure-core.TransportAdapter.connectionPoolTest ...........................***Failed  191.64 sec

```

**Review by katmsft — APPROVED** [review]

**Comment by ahsonkhan** [comment]
@vhvb1989 - why is the connection pool test failing?
https://dev.azure.com/azure-sdk/public/_build/results?buildId=611333&view=logs&j=5f43df56-f144-584b-8cac-2916b6897a0a&t=a63c70f6-59e4-531a-efba-82f28f72650d&l=2067

```text
71: Test command: D:\a\1\s\build\sdk\core\azure-core\test\ut\Debug\azure-core-test.exe "--gtest_filter=TransportAdapter.connectionPoolTest"
71: Test timeout computed to be: 10000000
71: Note: Google Test filter = TransportAdapter.connectionPoolTest
71: [==========] Running 1 test from 1 test suite.
71: [----------] Global test environment set-up.
71: [----------] 1 test from TransportAdapter
71: [ RUN      ] TransportAdapter.connectionPoolTest
71: Running Connection Pool Cleaner Test. This test takes more than 3 minutes to complete.
71: Add compiler option -DRUN_LONG_UNIT_TESTS=OFF when building if you want to skip this test.
71: First wait time done. Validating state.
71: D:\a\1\s\sdk\core\azure-core\test\ut\transport_adapter.cpp(98): error: Expected equality of these values:
71:   Http::CurlConnectionPool::ConnectionsOnPool("httpbin.org")
71:     Which is: 1
71:   2
71: [  FAILED  ] TransportAdapter.connectionPoolTest (191604 ms)
71: [----------] 1 test from TransportAdapter (191604 ms total
...
The following tests FAILED:
	 71 - azure-core.TransportAdapter.connectionPoolTest (Failed)
```

/azp run cpp - storage

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp [hunk 1]
```diff
@@ -53,127 +55,129 @@ namespace Azure { namespace Storage { namespace Blobs {
...
+      }
+      if (access_tier == "Archive")
+      {
+        return AccessTier::Archive;
+      }
+      throw std::runtime_error("cannot convert " + access_tier + " to AccessTier");
     }
-    throw std::runtime_error("cannot convert " + access_tier + " to AccessTier");
-  }
+  } // namespace Details
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:180]: Can you restructure/reorder the types in the file so you don't need to keep interspersing so many `Details` namespace in the middle?

File: sdk/storage/azure-storage-blobs/src/blob_sas_builder.cpp [hunk 2]
```diff
@@ -134,53 +134,58 @@ namespace Azure { namespace Storage { namespace Blobs {
         + resource + "\n" + snapshotVersion + "\n" + CacheControl + "\n" + ContentDisposition + "\n"
         + ContentEncoding + "\n" + ContentLanguage + "\n" + ContentType;
 
-    std::string signature
-        = Base64Encode(Details::HmacSha256(stringToSign, Base64Decode(credential.GetAccountKey())));
+    std::string signature = Base64Encode(
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/src/blob_sas_builder.cpp:137]: What's the plan for things in storage-common like `Base64Encode`. Such utility types shouldn't be public and probably belong to `::Details` as well.

**Comment by ahsonkhan** [comment]
@Jinming-Hu - pull from master to get CI to be green. It is currently failing because one of the links had their cert expire and now is invalid. @vhvb1989 removed it here: https://github.com/Azure/azure-sdk-for-cpp/pull/935

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:180]: it's not easy to do for the current auto generator.

anyway, this generator is supposed to be a temporary thing, and core sdk team is supposed to provide us with a full-fledged generator.

**Comment by Jinming-Hu** [comment]
> @Jinming-Hu - pull from master to get CI to be green. It is currently failing because one of the links had their cert expire and now is invalid. @vhvb1989 removed it here: #935

customers still want to use base64 encoding/decoding functions for MD5/CRC64

File: sdk/storage/azure-storage-blobs/src/blob_sas_builder.cpp (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/src/blob_sas_builder.cpp:137]: customers still want to use base64 encoding/decoding functions for MD5/CRC64

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/src/blob_sas_builder.cpp:137]: So the plan is we don't

**Pull request merged**